### PR TITLE
Client resource reload stuff

### PR DESCRIPTION
**Pull request by apple502j**
todo: better naming and packages?

### CODE REVIEW

File: mappings/net/minecraft/client/ReloadLogger.mapping [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	CLASS class_6361 RecoveryEntry
+		FIELD field_33701 throwable Ljava/lang/Throwable;
+		METHOD <init> (Ljava/lang/Throwable;)V
+			ARG 1 throwable
+		METHOD method_36567 addRecoverySection (Lnet/minecraft/class_129;)V
+			ARG 1 section
+	CLASS class_6362 ReloadReason
+		FIELD field_33705 reason Ljava/lang/String;
+		METHOD <init> (Ljava/lang/String;ILjava/lang/String;)V
+			ARG 3 reason
```
Comment (enbrain) [inline mappings/net/minecraft/client/ReloadLogger.mapping]:
```suggestion
		FIELD field_33705 name Ljava/lang/String;
		METHOD <init> (Ljava/lang/String;ILjava/lang/String;)V
			ARG 3 name
```
It's confusing to have code like `reloadReason.reason`.

File: mappings/net/minecraft/client/ReloadLogger.mapping [hunk 2]
```diff
@@ -0,0 +1,32 @@
+CLASS net/minecraft/class_6360 net/minecraft/client/ReloadLogger
+	FIELD field_33698 LOGGER Lorg/apache/logging/log4j/Logger;
+	FIELD field_33699 reload Lnet/minecraft/class_6360$class_6363;
+	FIELD field_33700 reloadNumber I
```
Comment (enbrain) [inline mappings/net/minecraft/client/ReloadLogger.mapping]:
```suggestion
	FIELD field_33700 reloadCount I
```
`number` is too abstract imo.

File: mappings/net/minecraft/client/ReloadLogger.mapping [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	METHOD method_36562 setFinished ()V
+	METHOD method_36563 reload (Lnet/minecraft/class_6360$class_6362;Ljava/util/List;)V
+		ARG 1 reloadReason
```
Comment (enbrain) [inline mappings/net/minecraft/client/ReloadLogger.mapping]:
```suggestion
	METHOD method_36563 reload (Lnet/minecraft/class_6360$class_6362;Ljava/util/List;)V
		ARG 1 reason
```

File: mappings/net/minecraft/client/ReloadLogger.mapping [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+		METHOD <init> (Ljava/lang/Throwable;)V
+			ARG 1 throwable
+		METHOD method_36567 addRecoverySection (Lnet/minecraft/class_129;)V
+			ARG 1 section
+	CLASS class_6362 ReloadReason
+		FIELD field_33705 reason Ljava/lang/String;
+		METHOD <init> (Ljava/lang/String;ILjava/lang/String;)V
+			ARG 3 reason
+	CLASS class_6363 ReloadEntry
+		FIELD field_33707 reloadReason Lnet/minecraft/class_6360$class_6362;
```
Comment (enbrain) [inline mappings/net/minecraft/client/ReloadLogger.mapping]:
```suggestion
		FIELD field_33707 reason Lnet/minecraft/class_6360$class_6362;
```

File: mappings/net/minecraft/client/ReloadLogger.mapping [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+		FIELD field_33708 packs Ljava/util/List;
+		FIELD field_33709 recovery Lnet/minecraft/class_6360$class_6361;
+		FIELD field_33710 finished Z
+		METHOD <init> (Lnet/minecraft/class_6360$class_6362;Ljava/util/List;)V
+			ARG 1 reloadReason
```
Comment (enbrain) [inline mappings/net/minecraft/client/ReloadLogger.mapping]:
```suggestion
			ARG 1 reason
```

File: mappings/net/minecraft/client/ReloadLogger.mapping [hunk 6]
```diff
@@ -0,0 +1,32 @@
+CLASS net/minecraft/class_6360 net/minecraft/client/ReloadLogger
+	FIELD field_33698 LOGGER Lorg/apache/logging/log4j/Logger;
+	FIELD field_33699 reload Lnet/minecraft/class_6360$class_6363;
+	FIELD field_33700 reloadNumber I
+	METHOD method_36562 setFinished ()V
```
Comment (enbrain) [inline mappings/net/minecraft/client/ReloadLogger.mapping]:
```suggestion
	METHOD method_36562 finish ()V
```

File: mappings/net/minecraft/client/ReloadLogger.mapping [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+		FIELD field_33701 throwable Ljava/lang/Throwable;
+		METHOD <init> (Ljava/lang/Throwable;)V
+			ARG 1 throwable
+		METHOD method_36567 addRecoverySection (Lnet/minecraft/class_129;)V
+			ARG 1 section
+	CLASS class_6362 ReloadReason
+		FIELD field_33705 reason Ljava/lang/String;
+		METHOD <init> (Ljava/lang/String;ILjava/lang/String;)V
+			ARG 3 reason
+	CLASS class_6363 ReloadEntry
```
Comment (enbrain) [inline mappings/net/minecraft/client/ReloadLogger.mapping]: What about `ReloadState`? `finished` field suggests it's a sort of state.
```suggestion
	CLASS class_6363 ReloadState
```

File: mappings/net/minecraft/client/ReloadLogger.mapping [hunk 8]
```diff
@@ -0,0 +1,32 @@
+CLASS net/minecraft/class_6360 net/minecraft/client/ReloadLogger
+	FIELD field_33698 LOGGER Lorg/apache/logging/log4j/Logger;
+	FIELD field_33699 reload Lnet/minecraft/class_6360$class_6363;
```
Comment (enbrain) [inline mappings/net/minecraft/client/ReloadLogger.mapping]:
```suggestion
	FIELD field_33699 reloadState Lnet/minecraft/class_6360$class_6363;
```

File: mappings/net/minecraft/client/ReloadLogger.mapping [hunk 9]
```diff
@@ -0,0 +1,32 @@
+CLASS net/minecraft/class_6360 net/minecraft/client/ReloadLogger
```
Comment (enbrain) [inline mappings/net/minecraft/client/ReloadLogger.mapping]: Since this is specifically for resource pack, what about `ResourceReloadLogger`?

**Review by enbrain — CHANGES_REQUESTED** [review]

**Review by enbrain — APPROVED** [review]

**Review by Player3324 — APPROVED** [review]

**Pull request merged**